src/System.Text.Json/src/System/Text/Json/Reader/Utf8JsonReader.TryGet.cs

```diff
@@ -469,8 +469,46 @@ public bool TryGetDateTime(out DateTime value)
                 throw ThrowHelper.GetInvalidOperationException_ExpectedString(TokenType);
             }
 
-            ReadOnlySpan<byte> span = HasValueSequence ? ValueSequence.ToArray() : ValueSpan;
-            return JsonHelpers.TryParseAsISO(span, out value, out int bytesConsumed) && span.Length == bytesConsumed;
+            ReadOnlySpan<byte> span = stackalloc byte[0];
+
+            if (HasValueSequence)
+            {
+                int sequenceLength = (int)ValueSequence.Length;
```

Comment:
nit: Should this be a long?

If `ValueSequence.Length > int.MaxValue32`, we want to return false (the `IsValidDateTimeOffsetParseLength` would catch it).

We should cast to int after `IsValidDateTimeOffsetParseLength` check is done (i.e. when we are stackallocating).

```C#
	{
		long x = (long)int.MaxValue + (long)int.MaxValue + (long)int.MaxValue;
		int sequenceLength = (int)x;
		Console.WriteLine(sequenceLength); // 2147483645
	}

	{
		long x = (long)int.MaxValue + 1L;
		int sequenceLength = (int)x;
		Console.WriteLine(sequenceLength); // -2147483648
	}
```

Otherwise, lets add a comment explaining why this cast is OK since `IsValidDateTimeOffsetParseLength` will return false if `sequenceLength > int.MaxValue` anyway.

Reply:
Changed to a `long`.